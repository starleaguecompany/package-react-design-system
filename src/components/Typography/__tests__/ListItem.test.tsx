import * as React from 'react'
import { render, screen } from '@testing-library/react'

import { ListItem } from '../index'
import { Icon } from '../../../index'
import { Menu } from '@starleaguecompany/react-icons'

describe('ListItem', () => {
  test('should render correctly with no props', () => {
    const { container } = render(<ListItem />)

    expect(container.firstChild).toMatchSnapshot()
    expect(container.firstChild).toHaveClass('item', { exact: true })
  })

  test('should render correctly with text', () => {
    render(<ListItem>Testing list item</ListItem>)

    expect(screen.getByRole('listitem')).toMatchSnapshot()
    expect(screen.getByRole('listitem')).toBeVisible()
    expect(screen.getByRole('listitem')).toHaveTextContent('Testing list item')
  })

  test('should render correctly with attributes', () => {
    render(
      <ListItem id="test-id" className="test-class">
        Testing list item
      </ListItem>
    )

    expect(screen.getByRole('listitem')).toMatchSnapshot()
    expect(screen.getByRole('listitem')).toHaveAttribute('id', 'test-id')
    expect(screen.getByRole('listitem')).toHaveClass('test-class')
  })

  test('should render correctly with icon', () => {
    render(<ListItem icon={<Icon icon={<Menu />} />}>Testing list item</ListItem>)

    expect(screen.getByRole('listitem')).toMatchSnapshot()
    expect(screen.getByTestId('Menu')).toBeVisible()
    expect(screen.getByRole('listitem')).toContainElement(screen.getByTestId('Menu'))
  })
})
