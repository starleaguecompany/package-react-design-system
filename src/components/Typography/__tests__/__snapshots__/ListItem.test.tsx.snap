// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem should render correctly with attributes 1`] = `
<li
  class="test-class item"
  id="test-id"
>
  Testing list item
</li>
`;

exports[`ListItem should render correctly with icon 1`] = `
<li
  class="item with-icon"
>
  <span
    class="icon"
  >
    <span
      class="container size-20 shape-round"
      data-qa="Icon"
    >
      <svg
        data-qa="Menu"
        fill="none"
        viewBox="0 0 20 20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M4 5h12M4 10h12M4 15h12"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="1.5"
        />
      </svg>
    </span>
  </span>
  Testing list item
</li>
`;

exports[`ListItem should render correctly with no props 1`] = `
<li
  class="item"
/>
`;

exports[`ListItem should render correctly with text 1`] = `
<li
  class="item"
>
  Testing list item
</li>
`;
